From 3ca45600a6cda653fa754e1b1081bc010063fe73 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Sat, 27 Nov 2021 17:02:44 +0100
Subject: [PATCH] Fix cmdfuel ruby script generation on Windows with MSVC

Signed-off-by: Silvio <silvio@traversaro.it>
---
 src/cmd/CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/cmd/CMakeLists.txt b/src/cmd/CMakeLists.txt
index 083a758..7b039fc 100644
--- a/src/cmd/CMakeLists.txt
+++ b/src/cmd/CMakeLists.txt
@@ -3,6 +3,8 @@
 # Ex: cmdfuel0.rb
 if (APPLE)
   set(IGN_LIBRARY_NAME lib${PROJECT_NAME_LOWER}.dylib)
+elseif(MSVC)
+  set(IGN_LIBRARY_NAME ${PROJECT_NAME_LOWER}.dll)
 else()
   set(IGN_LIBRARY_NAME lib${PROJECT_NAME_LOWER}.so)
 endif()
